### DISCUSSION TITLE
hyperledger fabric sdk java version 1.4

### DISCUSSION THREAD

**Discussion by Choi-Seong-Yong** [hyperledger/fabric-sdk-java#302] [Q&A]
In Hyperledger Fabric 1.4 version, node sdk supports offline signature, but is offline signature not supported in Java version?

**Comment by bestbeforetoday** [comment]
The legacy SDKs (include Node SDK and Java SDKs) are deprecated and no longer actively maintained. The only currently supported API for building Fabric client applications is the [Fabric Gateway client API](https://hyperledger.github.io/fabric-gateway/) (for Fabric v2.4 and later). This supports off-line signed for Go, Node (TypeScript/JavaScript), Java applications.
